Infrastructure.DataAccess/Migrations/SQLScripts/Deactive_Expired_Advice_And_Schedule_Missing_Jobs.sql

```diff
@@ -0,0 +1,217 @@
+/*
+User story reference: 
+    https://os2web.atlassian.net/browse/KITOSUDV-2243
+ 
+Content:
+	- Removes old Hangfire jobs which have no related data and should be deleted.
+	- Updates IsActive column on Advice, which should be deactivated when Hangfire runs next send. But won't be as there is no Hangfire job
+	- Recreates Hangfire jobs on Advices where there should be a job but isn't one.
+*/
+
+/* Note: Check to see if Hangfire tables exists so that script doesn't run on fresh install */
+IF OBJECT_ID('kitos_HangfireDB.HangFire.Set', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.Hash', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.Job', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.JobParameter', 'U') IS NOT NULL
+
+	BEGIN
+		/* Part 1: Delete recurring jobs with no related data from hangfire DB to clean up*/
+		DECLARE @RecurringJobsWithData TABLE 
+		(
+			AdviceId int
+		)
+
+		INSERT INTO @RecurringJobsWithData
+		SELECT CONVERT(int, SUBSTRING([Key], 23, 28)) AS AdviceId
+		FROM [kitos_HangfireDB].[HangFire].[Hash]
+		WHERE [Key] LIKE '%recurring-job:Advice: %'
+
+		INSERT INTO @RecurringJobsWithData
+		SELECT CONVERT(int, REVERSE(SUBSTRING(REVERSE(SUBSTRING([Arguments], 3, 8)), 3, 8))) AS AdviceId
+		FROM [kitos_HangfireDB].[HangFire].[Job]
+		WHERE [Arguments] NOT LIKE '%null%'
+
+
+		DECLARE @RecurringAdviceJobs TABLE 
+		(
+			AdviceId int,
+			SetId int
+		)
+
+		INSERT INTO @RecurringAdviceJobs
+		SELECT CONVERT(int, SUBSTRING([Value], 8, 6)), Id
+		FROM [kitos_HangfireDB].[HangFire].[Set]
+		WHERE [Key] LIKE 'recurring-jobs' AND [Score] > 0 AND [Value] LIKE 'Advice: %'
+
+		DELETE
+		FROM [kitos_HangfireDB].[HangFire].[Set] 
+		WHERE Id IN (
+			SELECT SetId
+			FROM @RecurringAdviceJobs
+			WHERE AdviceId NOT IN (
```

Comment:
har vi rent faktisk nogen hvor der er "set" data men ingen job data?

Reply:
i så fald er det skummelt, men for man skulle da tro at HF kørte sine egne ting i transaktioner

Reply:
Har ikke lige fået testet den del. Det her var mere for at være sikker på vi havde info fra begge steder. Jeg tester DB.

Reply:
Vi har 173 hvor det er tilfældet (De ser ret døde ud i HF) 
![image](https://user-images.githubusercontent.com/39791946/134860076-25f4bea6-aa87-41ce-9bb0-98f6b822b32b.png)